sklearn/metrics/classification.py

```diff
@@ -1911,9 +1911,29 @@ def brier_score_loss(y_true, y_prob, sample_weight=None, pos_label=None):
     y_prob = column_or_1d(y_prob)
     assert_all_finite(y_true)
     assert_all_finite(y_prob)
+    check_consistent_length(y_true, y_prob)
 
+    if y_prob.max() > 1:
+        raise ValueError("y_prob contains values greater than 1.")
+    if y_prob.min() < 0:
+        raise ValueError("y_prob contains values less than 0.")
+
+    # currently, we only support binary classification
+    classes = np.unique(y_true)
+    if len(classes) > 2:
+        raise ValueError("Only binary classification is supported.")
+
+    # ensure valid y_true if pos_label is not specified
     if pos_label is None:
-        pos_label = y_true.max()
+        if (np.array_equal(classes, [0]) or
+           np.array_equal(classes, [-1]) or
+           np.array_equal(classes, [1])):
+            pos_label = 1.
```

Comment:
@amueller Thanks.
Currently, log_loss don't support pos_label so the code is taken from _binary_clf_curve. But in the test we have to support situations when y_pred = ['egg', 'spam'] without a pos_label (see test_invariance_string_vs_numbers_labels). In order not to remove test, I give this awkward solution.
Btw, it might be OK not to use float like _binary_clf_curve because `np.array_equal([1.0], [1])=True`